app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -504,6 +525,7 @@ class LocationSensorManager : LocationSensorManagerBase() {
     }
 
     private fun removeBackgroundUpdateRequests() {
+        isBackgroundLocationSetup = false
```

Comment:
Won't this create problems when this function is called from [here](https://github.com/home-assistant/android/blob/90a9a27e5e371445f75481e7a93609528357d3ff/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L266)? `startHighAccuracyService(int)`, which is called in the next line, doesn't set it back to `true`.

Reply:
It should actually hit this part first

https://github.com/home-assistant/android/blob/90a9a27e5e371445f75481e7a93609528357d3ff/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L249-L255

Reply:
How would it end up in `setupBackgroundLocation` again? It isn't called from this function or the code where I linked to.

Reply:
Its actually just after the time evaluation I added, as part of `setupLocationTracking`

Reply:
I linked to `removeBackgroundUpdateRequests` from deeper inside `setupBackgroundLocation` when it is already setup and switching high accuracy mode from off -> on, so not in `setupLocationTracking` and after the "it should actually hit this part first" lines you linked to :)

Reply:
ok I see what you mean, will keep the variable logic within the condition :)